library/cc/engine_builder.cc

```diff
@@ -104,29 +104,29 @@ EngineBuilder& EngineBuilder::addStatsFlushSeconds(int stats_flush_seconds) {
   return *this;
 }
 
-EngineBuilder& EngineBuilder::addVirtualClusters(const std::string& virtual_clusters) {
-  this->virtual_clusters_ = virtual_clusters;
+EngineBuilder& EngineBuilder::addVirtualClusters(std::string virtual_clusters) {
+  this->virtual_clusters_ = std::move(virtual_clusters);
   return *this;
 }
 
-EngineBuilder& EngineBuilder::addKeyValueStore(const std::string& name,
+EngineBuilder& EngineBuilder::addKeyValueStore(std::string name,
                                                KeyValueStoreSharedPtr key_value_store) {
-  this->key_value_stores_[name] = key_value_store;
+  this->key_value_stores_[std::move(name)] = key_value_store;
```

Comment:
The key_value_store (which is a std::shared_ptr) should be moved to avoid a temporary shared_ptr (which results in an unnecessary atomic increment and immediate decrement).

Reply:
Done.